kolibri/core/analytics/utils.py

```diff
@@ -62,6 +73,133 @@
 ]
 
 
+def variance(data):
+    if data:
+        # calculate mean
+        m = sum(data) / len(data)
+        # return variance
+        return sum((d - m) ** 2 for d in data) / len(data)
+    return None
+
+
+def calculate_demographic_stats(dataset_id=None, channel_id=None, learner=True):
+    stats = {}
+    roles_filter = Q(roles__isnull=False)
+    queryset = FacilityUser.objects.all()
+    if learner:
+        roles_filter = Q(roles__isnull=True)
+    # handle stats at facility level
+    if dataset_id:
+        queryset = queryset.filter(roles_filter, dataset_id=dataset_id)
+    # handle stats at channel level
+    if channel_id:
+        queryset = queryset.filter(
+            roles_filter, contentsummarylog__channel_id=channel_id
```

Comment:
Could you not just filter by the `roles_filter` once rather than doing it in each conditional?